configserver/src/main/java/com/yahoo/vespa/config/server/session/RemoteSession.java

```diff
@@ -26,95 +15,46 @@
+     * Creates a remote session, no application set loaded
      *
      * @param tenant The name of the tenant creating session
      * @param sessionId The session id for this session.
-     * @param componentRegistry a registry of global components
      * @param zooKeeperClient a SessionZooKeeperClient instance
      */
     public RemoteSession(TenantName tenant,
                          long sessionId,
-                         GlobalComponentRegistry componentRegistry,
                          SessionZooKeeperClient zooKeeperClient) {
-        super(tenant, sessionId, zooKeeperClient);
-        this.applicationLoader = new ActivatedModelsBuilder(tenant, sessionId, zooKeeperClient, componentRegistry);
-        this.clock = componentRegistry.getClock();
-    }
-
-    void prepare() {
-        Curator.CompletionWaiter waiter = sessionZooKeeperClient.getPrepareWaiter();
-        ensureApplicationLoaded();
-        notifyCompletion(waiter);
+        this(tenant, sessionId, zooKeeperClient, Optional.empty());
     }
 
-    private ApplicationSet loadApplication() {
-        ApplicationPackage applicationPackage = sessionZooKeeperClient.loadApplicationPackage();
-
-        // Read hosts allocated on the config server instance which created this
-        Optional<AllocatedHosts> allocatedHosts = applicationPackage.getAllocatedHosts();
-
-        return ApplicationSet.fromList(applicationLoader.buildModels(getApplicationId(),
-                                                                     sessionZooKeeperClient.readDockerImageRepository(),
-                                                                     sessionZooKeeperClient.readVespaVersion(),
-                                                                     applicationPackage,
-                                                                     new SettableOptional<>(allocatedHosts),
-                                                                     clock.instant()));
+    /**
+     * Creates a remote session, with application set
+     *
+     * @param tenant The name of the tenant creating session
+     * @param sessionId The session id for this session.
+     * @param zooKeeperClient a SessionZooKeeperClient instance
+     */
+    public RemoteSession(TenantName tenant,
+                         long sessionId,
+                         SessionZooKeeperClient zooKeeperClient,
+                         Optional<ApplicationSet> applicationSet) {
+        super(tenant, sessionId, zooKeeperClient);
+        this.applicationSet = applicationSet;
     }
 
-    public synchronized ApplicationSet ensureApplicationLoaded() {
-        return applicationSet == null ? applicationSet = loadApplication() : applicationSet;
+    Optional<ApplicationSet> applicationSet() {
+        return applicationSet;
     }
 
-    public synchronized void deactivate() {
-        applicationSet = null;
+    public synchronized RemoteSession deactivate() {
```

Comment:
Would like a name change here, to `deactivated()`

Reply:
👍